### PR TITLE
Javascript browser drawing options

### PR DESCRIPTION
**Pull request by Falcort**
Added the functionality to have drawings options per graph types on the javascript browser

### CODE REVIEW

File: gui/browserv7/inc/ROOT/RBrowser.hxx [hunk 1]
```diff
@@ -96,7 +97,7 @@ protected:
    std::string GetClassIcon(std::string &classname);
    std::string GetFileIcon(TString &name);
    std::string ProcessBrowserRequest(const std::string &msg);
-   std::string ProcessDblClick(const std::string &path);
+   std::string ProcessDblClick(const std::string &path, const std::string drawingOptions);
```
Comment (linev) [inline gui/browserv7/inc/ROOT/RBrowser.hxx]: You always should use `const std::string &` to pass string parameter

File: gui/browserv7/src/RBrowser.cxx [hunk 2]
```diff
@@ -391,8 +392,7 @@ std::string ROOT::Experimental::RBrowser::ProcessBrowserRequest(const std::strin
 /////////////////////////////////////////////////////////////////////////////////
 /// Process dbl click on browser item
 
-std::string ROOT::Experimental::RBrowser::ProcessDblClick(const std::string &item_path)
-{
+std::string ROOT::Experimental::RBrowser::ProcessDblClick(const std::string &item_path, const std::string drawingOptions) {
    if (item_path.find(".root") == std::string::npos) {
```
Comment (linev) [inline gui/browserv7/src/RBrowser.cxx]: same of course here

File: gui/browserv7/src/RBrowser.cxx [hunk 3]
```diff
@@ -464,13 +464,13 @@ std::string ROOT::Experimental::RBrowser::ProcessDblClick(const std::string &ite
...
-      TH1 *h1 = dynamic_cast<TH1*>(clone);
+      TObject *clone = object->Clone();
+      TH1 *h1 = dynamic_cast<TH1 *>(clone);
       if (h1) h1->SetDirectory(nullptr);
 
       std::shared_ptr<TObject> ptr;
       ptr.reset(clone);
-      rcanv->Draw<RObjectDrawable>(ptr, "");
+      rcanv->Draw<RObjectDrawable>(ptr, drawingOptions.c_str());
       rcanv->Modified();
```
Comment (linev) [inline gui/browserv7/src/RBrowser.cxx]: Here no need to use c_str() - just provide `drawingOptions`

File: gui/browserv7/src/RBrowser.cxx [hunk 4]
```diff
@@ -693,8 +693,21 @@ void ROOT::Experimental::RBrowser::WebWindowCallback(unsigned connid, const std:
 
       fWebWindow->Send(connid, res);
    } else if (arg.compare(0,7, "DBLCLK:") == 0) {
-      auto str = ProcessDblClick(arg.substr(7));
-      if (str.length() > 0) fWebWindow->Send(connid, str);
+
+      if (arg.at(8) != '[') {
```
Comment (linev) [inline gui/browserv7/src/RBrowser.cxx]: Is is necessary here? 
Do you have situation when options not send and just itemname is provided?

File: ui5/browser/controller/Browser.controller.js [hunk 5]
```diff
@@ -214,12 +229,42 @@ sap.ui.define(['sap/ui/core/Component',
...
+       if (!this._oSettingsMenu) {
+         let myThis = this;
+         await sap.ui.core.Fragment.load({name: "rootui5.browser.view.settingsmenu"}).then(function (oSettingsMenu) {
+           let oModel = new sap.ui.model.json.JSONModel("rootui5sys/browser/json/drawingOptions.json");
+           oSettingsMenu.setModel(oModel);
+           oSettingsMenu.attachConfirm(myThis.handleSettingsConfirm);
+           myThis.getView().addDependent(oSettingsMenu);
+
+           myThis._oSettingsMenu = oSettingsMenu;
+           return oSettingsMenu;
```
Comment (linev) [inline ui5/browser/controller/Browser.controller.js]: I guess, this return is useless here

File: ui5/browser/controller/Browser.controller.js [hunk 6]
```diff
@@ -214,12 +229,42 @@ sap.ui.define(['sap/ui/core/Component',
...
          var file = oEvent.getParameter("files")[0];
          if (this.setFileNameType(file.name))
             oReader.readAsText(file);
       },
 
+     _getSettingsMenu: async function () {
+       if (!this._oSettingsMenu) {
+         let myThis = this;
+         await sap.ui.core.Fragment.load({name: "rootui5.browser.view.settingsmenu"}).then(function (oSettingsMenu) {
+           let oModel = new sap.ui.model.json.JSONModel("rootui5sys/browser/json/drawingOptions.json");
```
Comment (linev) [inline ui5/browser/controller/Browser.controller.js]: Maybe we can avoid creation of extra JSON file and just put JSON into controller?
like:
```
new JSONModel({"TH1": [
		{
			"name": "hist"
		},
		...
})
```

Comment (linev) [reply ui5/browser/controller/Browser.controller.js]: Also you should avoid use of fully-qualified class names like `sap.ui.model.json.JSONModel`.
They should be obtained via `sap.ui.define` (actually JSONModel is there)

Comment (Falcort) [reply ui5/browser/controller/Browser.controller.js]: I mean i can, but the JSON is actually 200lines long, and only have TH1, TH2 and TProfile, if we add more, i will grow quite a lot, that is why i made it into a separate file.
But sure i can do it if you want

File: ui5/browser/controller/Browser.controller.js [hunk 7]
```diff
@@ -267,6 +336,15 @@ sap.ui.define(['sap/ui/core/Component',
             return this.websocket.Send("DBLCLK:" + fullpath);
        },
 
+      getMasterClass: function(className) {
```
Comment (bellenot) [inline ui5/browser/controller/Browser.controller.js]: I would prefer `getBaseClass` as method name...

File: ui5/browser/controller/Browser.controller.js [hunk 8]
```diff
@@ -124,6 +126,16 @@ sap.ui.define(['sap/ui/core/Component',
...
+            t.addColumn(new tableColumn({
+              label: "ClassName",
+              autoResizable: true,
+              visible: false,
+              template: new HorizontalLayout({
+                content: [
+                  new mText({text:"{className}", wrapping: false })
+                ]
+              })
+            }));
```
Comment (bellenot) [inline ui5/browser/controller/Browser.controller.js]: Can't we re-use an existing one?

File: gui/browserv7/src/RBrowser.cxx (hunk 4 above)
Comment (Falcort) [reply gui/browserv7/src/RBrowser.cxx]: I think so, without it an editor request which look like this : 
`DBLCLK:ROOTConfig.cmake`
and a root request which look like : 
`DBLCLK: ["tutorials/hsimple.root/hpx;1","hist"]`
so on the line :
`auto arr = TBufferJSON::FromJSON<std::vector<std::string>>(arg.substr(7));`
With the editor there isn't an array, or even a drawing option

File: gui/browserv7/src/RBrowser.cxx (hunk 3 above)
Comment (Falcort) [reply gui/browserv7/src/RBrowser.cxx]: So i should remove it here too right ?
`canv->GetListOfPrimitives()->Add(object, drawingOptions.c_str());`

File: gui/browserv7/src/RBrowser.cxx (hunk 4 above)
Comment (linev) [reply gui/browserv7/src/RBrowser.cxx]: Agree, dblclick may be handled in other situations too

File: gui/browserv7/src/RBrowser.cxx (hunk 3 above)
Comment (Falcort) [reply gui/browserv7/src/RBrowser.cxx]: ok, thanks

**Comment by Falcort** [comment]
@linev @bellenot Everything should be resolve in 99d936c

**Pull request merged**